impl_unix.go

```diff
@@ -157,6 +157,33 @@ func dial(ctx context.Context, dialer net.Dialer, netw, addr string) (c net.Conn
 	}
 
 	if rprotocol == unix.IPPROTO_TCP {
+		sa, err := unix.Getsockname(fd)
+		if err != nil {
+			unix.Close(fd)
+			return nil, err
+		}
+		ra, err := unix.Getpeername(fd)
+		if err != nil {
+			unix.Close(fd)
+			return nil, err
+		}
+
+		// Need to call setLinger 0. Otherwise, the close will block for the linger period. Linux bug?
```

Comment:
more like extreme in-spec behaviour i would say.

Reply:
`man 3 close`:

> If fildes refers to a socket, close() shall cause the socket to be destroyed. If the socket is in connection-mode, and the SO_LINGER option is set for the socket with non-zero linger time, and the socket has untransmitted data, then close() shall block for up to the current linger interval until all data is transmitted.

However, we *don't* have untransmitted data (and no, reading from the socket in a different thread doesn't prevent close from blocking).